builder/builder.go

```diff
@@ -217,11 +211,21 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 		// to accomodate parallelism in the WAIT/END PopWaitBlock handling
 		dirIDs = map[int]string{}
 	)
+
 	var (
 		depIndex   = 0
 		imageIndex = 0
 		dirIndex   = 0
 	)
+
+	// Delay closing the registry proxy server until this method
+	// returns. Previously, it was being closed when the buildFunc returned,
```

Comment:
nit: no need to mention a past bug that was fixed in this comment.

I would reword this as simply: `Delay closing the registry proxy server until after the buildFunc returns`

Reply:
Sure. I wanted to allude to the fact that this code can be reverted once global wait block support is enabled. Updated.